launch/launch/conditions/launch_configuration_equals.py

```diff
@@ -43,6 +44,16 @@ def __init__(
         launch_configuration_name: Text,
         expected_value: Optional[SomeSubstitutionsType]
     ) -> None:
+        warnings.warn(
```

Comment:
I would also add a notice to the docstring above.

Reply:
I added a Sphinx deprecation directive